### ISSUE TITLE
Upgrade to Sauce Connect 5 and support `--proxy-localhost`

### ISSUE THREAD

**Issue by GraemeWatt** [saucelabs/sauce-connect-action#88]
It seems this action only works with Sauce Connect 4, where support will be discontinued as of December 31, 2024.  Sauce Connect 4 will reach end of life on May 5, 2025, after which no tunnels will be available.  Are there plans to upgrade this action to work with Sauce Connect 5?  I can see that the download URLs are a different format in Sauce Connect 4 and Sauce Connect 5.

I also need to use the [`--proxy-localhost`](https://docs.saucelabs.com/dev/cli/sauce-connect-5/run/#proxy-localhost) flag with Sauce Connect 5.  This is also [available](https://docs.saucelabs.com/dev/cli/sauce-connect-proxy/#--proxy-localhost) in Sauce Connect 4, but it does not seem supported by `sauce-connect-action`.

**Comment by alexh-sauce** [comment]
Yes, we're working on upgrading all of the plugins that are SC4 only to SC5, and will make sure that's done before SC4 is EOL'd.
